include/parthenon/CMakeLists.txt

```diff
@@ -0,0 +1,81 @@
+#=========================================================================================
+# (C) (or copyright) 2020. Triad National Security, LLC. All rights reserved.
+#
+# This program was produced under U.S. Government contract 89233218CNA000001 for Los
+# Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+# for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+# in the program are reserved by Triad National Security, LLC, and the U.S. Department
+# of Energy/National Nuclear Security Administration. The Government is granted for
+# itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+# license in this material to reproduce, prepare derivative works, distribute copies to
+# the public, perform publicly and display publicly, and to permit others to do so.
+#=========================================================================================
+
+# The header files are added to parthenon for linting
+list(APPEND PARTHENON_PUBLIC_HEADERS
```

Comment:
Hm... So what I typically like to do in a cmake project is:

```
CMakeLists.txt
src/
    *.cpp
include/
    *.hpp
```

And then the `add_library` goes into the CMakeLists.txt that sits next to `src`. It would resolve having to do this thing you're doing here with collecting the header files in a separate CMakeLists.txt.

So, should we perhaps just eliminate `src/CMakeLists.txt` and move its contents into the parent directory?

Reply:
Or maybe add a `parthenon` subdirectory and then move both src and include into that directory if you want to keep it cleanly separated from the other files in the root directory.

Reply:
I think it would be a good idea to keep the current directory structure because it follows convention. However, I'm up for suggestions. I think the only reason a list of the header files are needed is because of the way the linter is set, that might be the best spot to look into changing things.